R/fread.R

```diff
@@ -295,7 +295,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir())
              "complex" = as.complex(v),
              "raw" = as_raw(v),  # Internal implementation
              "Date" = as.Date(v),
-             "POSIXct" = as.POSIXct(v),
+             # "POSIXct" = as.POSIXct(v),  # v1.13.0 now reads POSIXct directly, #4464 
```

Comment:
doesn't that mean it will dispatch to `methods::as(v, new_class)`?

Reply:
but it's already a POSIXct so as() will just return `v` straight back.  As with any explicit class where fread has already read that class directly.

Reply:
on second thought, why was the as.POSIXct there in the first place; i.e. why didn't we leave it to `as(v, new_class)`? I seem to remember `as(v, "POSIXct")` is slower than `as.POSIXct()`. Perhaps to avoid method lookup.
If we did add the line back and cover it, iiuc, we'd need to construct a datetime field that `fread` did not convert to `POSIXct` directly, but could still be converted with `as.POSIXct`. Or, maybe a date column which fread reads directly as `IDate`, but user explicitly passes `colClasses=POSIXct` intending a date-only stored as POSIXct.

Reply:
cc @HughParsonage , I seem to recall your PR started with using `as(., class)`, any insight on using `as.POSIXct` directly?

Reply:
It simply doesn't work: 

``` r
x <- "2018-05-01 15:55:01"
as(x, "POSIXct")
#> Error in as(x, "POSIXct"): no method or default for coercing "character" to "POSIXct"
as.POSIXct(x)
#> [1] "2018-05-01 15:55:01 AEST"
```

Reply:
Thanks, @HughParsonage.
I tried putting back the as.POSIXct and covering it, but it turns out to be unreachable. For the reason that because POSIXct is now a recognized direct type, specifying `colClasses=POSIXct` for a date-only column (how I was trying to cover it), overrides the auto-detected IDate reader and bumps to use the POSIXct direct reader directly at C level before reading. (Which currently fails on date-only, and therefore bumps to character in out-of-sample, but that's a separate minor issue to fix which I added to the list at the top of this issue.)
So this switch should not contain any cases that have direct readers at C level. I'll add a comment about this to the code pointing here.

Reply:
resolved by eaf2d57